app-frontend/src/store/reducers/authorization.ts

```diff
@@ -0,0 +1,72 @@
+import initialState from '../initialState';
+import { UserInterface } from '../../types/user';
+
+enum ActionTypes {
+  AUTHORIZE_USER_SUCCESS = 'AUTHORIZE_USER_SUCCESS',
+  AUTHORIZE_USER_DATA = 'AUTHORIZE_USER_DATA',
+  AUTHORIZE_USER_FAILED = 'AUTHORIZE_USER_FAILED',
+  REGISTER_USER_SUCCESS = 'REGISTER_USER_SUCCESS',
+  REGISTER_USER_DATA = 'REGISTER_USER_DATA',
+  REGISTER_USER_FAILED = 'REGISTER_USER_FAILED',
+  GET_CURRENT_USER_DATA = 'GET_CURRENT_USER_DATA',
+  GET_CURRENT_USER_FAILED = 'GET_CURRENT_USER_FAILED',
+  GET_CURRENT_USER_SUCCESS = 'GET_CURRENT_USER_SUCCESS',
+}
+
+interface LoginUserAction {
+  type: ActionTypes.AUTHORIZE_USER_DATA
+  | ActionTypes.REGISTER_USER_DATA | ActionTypes.GET_CURRENT_USER_DATA,
+}
+
+interface LoginUserSuccessAction {
+  type: ActionTypes.AUTHORIZE_USER_SUCCESS
+  | ActionTypes.REGISTER_USER_SUCCESS | ActionTypes.GET_CURRENT_USER_SUCCESS,
+  payload: UserInterface,
+}
+
+interface LoginUserFailedAction {
+  type: ActionTypes.AUTHORIZE_USER_FAILED
+  | ActionTypes.REGISTER_USER_FAILED | ActionTypes.GET_CURRENT_USER_FAILED,
+  payload: string,
+}
+
+type UserAction = LoginUserAction | LoginUserSuccessAction | LoginUserFailedAction;
```

Comment:
All types of actions should be placed into action creators files, and action creators must be typed